### DISCUSSION TITLE
PageDesigner Beta : Default layout issues

### DISCUSSION THREAD

**Discussion by rchemk** [SalesforceCommerceCloud/sfcc-ci#430] [Team Posts]
Hello all,

My team and I are trying to implement the Page Designer (beta) on our sandbox, in order to make a demonstration to our customers.

We followed the documentation available on the Page Designer section of the BM (Getting Started - https://xchange.demandware.com/community/help_and_learning/content-and-content-management-resource-center)
(We installed the cartridges, added them on the path, and installed the default pages available for Sitegenesis).

Everything seemed to work fine, the 3 default pages configured worked perfectly (Images and Texts were OK).
But then we tried to create pages using the given layouts / blocs; 
This is when our issues appeared. From the configuration page, everything works: We created a page with only a Headline Banner component; With an image, a title and a secondary title, that looks like this
<img width="959" alt="headline_banner_config" src="https://user-images.githubusercontent.com/48559195/60874527-78cbc800-a238-11e9-992a-572d75176dc5.png">

But when we open the page from our storefront, the image doesn't appear, and the texts are "undefined" 
<img width="953" alt="undefined" src="https://user-images.githubusercontent.com/48559195/60874712-be889080-a238-11e9-830a-1541419cfcab.png">

An error appears in our error logs saying : 
`TypeError: Cannot read property "src" from null ([Template:experience/components/assets/headlinebanner:${pdict.image.src.mobile}]#1)
	at [Template:experience/components/assets/headlinebanner:${pdict.image.src.mobile}]:1`
and 
`TypeError: Cannot read property "alt" from null ([Template:experience/components/assets/headlinebanner:${pdict.image.alt}]#1)
	at [Template:experience/components/assets/headlinebanner:${pdict.image.alt}]:1`

I tried to debug the thing, but I can't find any solution atm.

Have you ever faced the same issue ? If so, how did you fix it ?

Thanks a lot.

**Comment by mlenzner** [comment]
I guess you have not setup all the component attributes in the locale you render the storefront for, i.e. created the page in locale 'default' only but then rendered in 'en_US' in the storefront so it misses all the attributes. Until 19.8 you have to provide the attributes separately for each required locale. With 19.8 locale fallback will be available, so that the above scenario would work out of the box. Please let us know if that is the issue you are facing.

**Comment by rchemk** [comment]
Hello @mlenzner,
Yes this is the issue, I didn't realize that the problem could come from the locales (I thought fallback locales worked everywhere by default).

Thanks a lot, that was a quick fix :D
